test/e2e/app-dir/app-prefetch/prefetching.test.ts

```diff
@@ -329,4 +329,22 @@ describe('app dir - prefetching', () => {
       })
     })
   })
+
+  describe('invalid URLs', () => {
+    it('should not throw when an invalid URL is passed to Link', async () => {
+      const browser = await next.browser('/invalid-url/from-link')
+
+      await check(() => browser.hasElementByCssSelector('h1'), true)
+      expect(await browser.elementByCss('h1').text()).toEqual('Hello, world!')
+    })
+
+    it('should throw when an invalid URL is passed to router.prefetch', async () => {
```

Comment:
We may want to change this later and make `router.prefetch` swallow the error. But this matches existing behavior, and this PR is intended to go into the v14.2 backport branch, so i feel like we may want to preserve what was there for now.